proposed/project-file-simplification.md

```diff
@@ -0,0 +1,351 @@
+
+### Upgrade tooling
+
+While the user can directly change the value specified in the `Base` attribute, this design allows a more sophisticated upgrade experience.
+
+We can determine what the project file would look like with the previous `Base` (or no `Base`) and with the proposed `Base`.See the section [Changing the project file, retaining behavior](#changing-the-project-file-retaining-behavior).
+
+The upgrade assistant should also be explored as part of this process.
+
+#### Including help information in the `Base`
+
+It would be desirable to include extra information as part of each property in the `Base` file. This could be used as part of an upgrade process, used in generating docs, passed on to users, or made available to AI. The desired information would be:
+
+| Information                             |
+|-----------------------------------------|
+| Description                             |
+| URL for more info                       |
+| SDK version where first added to `Base` |
+
+This information should be semantically available (possibly through MSBuild metadata) and also be helpful to humans perusing the file to see what's new.
+
+#### Changing the project file, retaining behavior
+
+One approach to the tooling would be to run a process that updates the current project to one that has identical behavior using the new `Base` by placing new properties into the project file. The user could then explore which of these to remove to enable new behavior. The user would start this process with a command like:
+
+```bash
+dotnet project upgrade 11.0.100
+```
+
+This would update the project file with the requested `Base` and new properties based on a _diff_ between the previous and resulting project properties.
+
+For example, if the 11.0.100 version of `Base` had two new enabled properties `UseFeatureA` and `UseFeatureB`, two new properties would be added to the project file when upgrading from a version 10 to version 11:
+
+```XML
+<Project Base="10.0.100" 
+         Type="Console"> 
+</Project> 
+```
+
+```XML
+<Project Base="11.0.100" 
+         Type="Console">
+    <PropertyGroup Label="These properties added on upgrade from 10.0.100 to 11.0.100"
+       <UseFeatureA>disable</UseFeatureA>
+       <UseFeatureB>disable</UseFeatureB>
+    </PropertyGroup>
+</Project> 
+```
+
+This approach is simple for a project file. It might be extended to complex repos by placing the `PropertyGroup` containing the new properties at the start of whichever file (project file, `Directory.Build.props` or `Import` file) the `Base` attribute appeared in. However, the command would probably be run in the same directory as the file containing the `Base` attribute and specify the file if there were multiple props files.
```

Comment:
```suggestion
This approach is simple for a project file. It might be extended to complex repos by placing the `PropertyGroup` containing the new properties at the start of whichever file (project file, `Directory.Build.props` or `Import` file) the `Base` attribute appeared in. However, the command would probably be run in the same directory as the file containing the `Base` attribute and specify the file if there were multiple `.props` files.
```